packages/compass-query-bar/src/stores/ai-query-reducer.ts

```diff
@@ -257,12 +255,13 @@ export const cancelAIQuery = (): QueryBarThunkAction<
   };
 };
 
-export const showInput = (): QueryBarThunkAction<void> => {
-  return (dispatch, getState) => {
-    if (getState().atlasSignIn.state === 'success') {
+export const showInput = (): QueryBarThunkAction<Promise<void>> => {
+  return async (dispatch, _getState, { atlasService }) => {
+    try {
+      await atlasService.signIn({ promptType: 'ai-promo-modal' });
```

Comment:
That's the only thing query-bar need to do now to ensure that user is signed in before showing the input. This also just immediately resolves when already signed in in other tabs, so works seamless across different instances of the plugin